### PR TITLE
Feature/request logger

### PR DESCRIPTION
**Pull request by remilvus**

### CODE REVIEW

File: prod-docker-compose.yml [hunk 1]
```diff
@@ -48,12 +48,13 @@ services:
...
-      - "origin:10.0.156.95" 
-      - "b.com:10.0.156.95" 
-      - "a.com:10.0.156.95" 
+      # change it so private IPs are not hard coded
+      - "origin:10.0.156.95"
+      - "b.com:10.0.156.95"
+      - "a.com:10.0.156.95"
     volumes:
       - /mnt/image-cache-part/cache:/data/cache
+      - /log/:/log/
```
Comment (Qizot) [inline prod-docker-compose.yml:57]: Can we store logs in `/mnt/image-cache-part/log`? I believe that unix has no such directory as `/log` by itself (and it is rather a bad practise to put some new folders under root directory).

File: src/main/java/com/io/image/manager/controller/ImageController.java [hunk 2]
```diff
@@ -22,15 +22,22 @@
...
 import java.util.List;
 import java.util.Optional;
+import java.util.stream.Collectors;
 
 @RestController
 public class ImageController {
     private final ImageService imageService;
     private final DistributionSummary outboundTrafficSummary;
     private final AppConfigurationProperties props;
+    private final String logPath = "/log/IM_log.txt";
```
Comment (Qizot) [inline src/main/java/com/io/image/manager/controller/ImageController.java:39]: Move that path to application config and access it from `AppConfigurationProperies` (the `props` field above)

File: src/main/java/com/io/image/manager/controller/ImageController.java [hunk 3]
```diff
@@ -83,14 +90,43 @@ public ImageController(ImageService imageService, PrometheusMeterRegistry mr, Ap
...
     }
 
     private OriginServer originFromHost(String host) {
         return new OriginServer(String.format("https://%s/", host));
     }
+
+    private void logRequest(String filename, String origin, boolean isPresent, List<ImageOperation> operations){
+        try {
+            if (writer == null){
+                writer = new BufferedWriter(new FileWriter(logPath, true));
```
Comment (Qizot) [inline src/main/java/com/io/image/manager/controller/ImageController.java:109]: Are you sure there won't be any race conditions? I don't know spring boot well and I have no idea if controllers are singletons or not, @Przemyslaw5 do you happen to know how it works?

File: src/main/java/com/io/image/manager/controller/ImageController.java [hunk 4]
```diff
@@ -83,14 +90,43 @@ public ImageController(ImageService imageService, PrometheusMeterRegistry mr, Ap
...
+        logRequest(filename, props.getOriginServer(), false, operations);
 
         throw new ResponseStatusException(HttpStatus.NOT_FOUND);
     }
 
     private OriginServer originFromHost(String host) {
         return new OriginServer(String.format("https://%s/", host));
     }
+
+    private void logRequest(String filename, String origin, boolean isPresent, List<ImageOperation> operations){
```
Comment (Qizot) [inline src/main/java/com/io/image/manager/controller/ImageController.java:106]: A comment explaining this log format would be helpful.

**Review by Qizot — CHANGES_REQUESTED** [review]

File: src/main/java/com/io/image/manager/controller/ImageController.java (hunk 3 above)
Comment (remilvus) [reply src/main/java/com/io/image/manager/controller/ImageController.java:109]: I'm not sure

File: prod-docker-compose.yml [hunk 5]
```diff
@@ -46,14 +46,16 @@ services:
...
-      - "origin:10.0.156.95" 
-      - "b.com:10.0.156.95" 
-      - "a.com:10.0.156.95" 
+      # change it so private IPs are not hard coded
+      - "origin:10.0.156.95"
+      - "b.com:10.0.156.95"
+      - "a.com:10.0.156.95"
     volumes:
       - /mnt/image-cache-part/cache:/data/cache
+      - /mnt/log:/data/log
```
Comment (Qizot) [inline prod-docker-compose.yml:58]:
```suggestion
      - /mnt/image-cache-part/log:/data/log
```

File: dev-docker-compose.yml [hunk 6]
```diff
@@ -54,6 +54,8 @@ services:
       - 8080:8080
     networks:
       - grafana
+    volumes:
+      - ./log/:/log/
```
Comment (Qizot) [inline dev-docker-compose.yml:58]: could you add `log` folder to `.gitignore` or at least check that we would not add it by accident to git?

**Review by Qizot — CHANGES_REQUESTED** [review]

File: dev-docker-compose.yml (hunk 6 above)
Comment (remilvus) [reply dev-docker-compose.yml:58]: There is already `/log/` in `.gitignore`

**Pull request merged**